### PR TITLE
Feature/orca 556 Implement internal_reconcile_report_phantom sql functionality into GraphQL prototype

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Added phantom retrieval to GraphQL

Addresses [ORCA-556: Implement internal_reconcile_report_phantom sql functionality into GraphQL prototype](https://bugs.earthdata.nasa.gov/browse/ORCA-556)

## Changes

* Added phantom retrieval to GraphQL
* Fixed bug with mismatch between GraphQL and Python definitions of int

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Retrieval run against ORCA database both locally and in AWS.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: graphql/src/entities/internal_reconcile_report.py [hunk 1]
```diff
@@ -7,22 +7,84 @@
 import strawberry
 
 
+@strawberry.type  # Not strictly clean, but alternative is duplicating classes in graphql adapter.
+@dataclasses.dataclass
+class Phantom(pydantic.BaseModel):
+    # IMPORTANT: Whenever properties are added/removed/modified/renamed, update constructor.
+    # Python doesn't cap 32 bit/4 byte int size, but GraphQL can't handle larger ints.
```
Comment (AndrewDorn) [inline graphql/src/entities/internal_reconcile_report.py:14]: This does unfortunately mean that these values, which should be integers, will be returned with a decimal (1234.0). Might be a way to define a custom GraphQL type, but it wasn't straightforward.

Comment (AndrewDorn) [reply graphql/src/entities/internal_reconcile_report.py:14]: https://bugs.earthdata.nasa.gov/browse/ORCA-622

**Review by rizbihassan — COMMENTED** [review]
looks good to me.

File: graphql/src/use_cases/adapter_interfaces/storage.py [hunk 2]
```diff
@@ -43,7 +70,7 @@ def get_mismatch_page(
         ordered by collection_id, granule_id, then key_path.
 
         Args:
-            job_id: The job to return mismatches for.
+            job_id: The job to return reports for.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply graphql/src/use_cases/adapter_interfaces/storage.py:73]: Eh. It's in the title, and the description. Trying to cut down on the differences between copy/paste functions wherever I can.

**Review by rizbihassan — APPROVED** [review]

**Pull request merged**